client/components/site-selector/index.jsx

```diff
@@ -351,6 +351,11 @@ export class SiteSelector extends Component {
 				isHighlighted={ isHighlighted }
 				isSelected={ this.isSelected( ALL_SITES ) }
 				title={ multiSiteContext && multiSiteContext.navigationLabel }
+				showCount={ false }
+				showIcon={ true }
+				icon={
+					<span className={ 'dashicons-before ' + multiSiteContext.icon } aria-hidden={ true } />
```

Comment:
That's cool! Any reason not to use https://github.com/Automattic/wp-calypso/blob/0d17396107f113bd675fb37ff0e0559817975045/client/layout/sidebar/custom-icon.jsx#L11?